go-controller/pkg/ovn/ovn.go

```diff
@@ -153,21 +151,20 @@ const (
 
 // NewOvnController creates a new OVN controller for creating logical network
 // infrastructure and policy
-func NewOvnController(kubeClient kubernetes.Interface, wf *factory.WatchFactory, stopChan <-chan struct{}) *Controller {
+func NewOvnController(kubeClient kubernetes.Interface, wf *factory.WatchFactory, stopChan <-chan struct{}, ipamFunc ipamFactoryFunc) *Controller {
```

Comment:
if the ipamFunc is just used for testing, it's acceptable to just overwrite private variables in test code. I wouldn't add it as another parameter.

Reply:
> @abhat some high level comments
> 
>     1. I think we need ipallocator's NewContiguousAllocationMap strategy which allocates the IP in contiguous manner like how @dcbw describes here. [#1365 (comment)](https://github.com/ovn-org/ovn-kubernetes/pull/1365#issuecomment-634045538)
> 
>     2. We tend to use `net.IP` instead of `*net.IP`, so to be consistent can you please change your usage of *net.IP to net.IP

I change the allocator to use the contiguous allocation strategy. But at any rate, if we are going to move the logic inside the IP allocator as @squeed suggested, we will have that be the default allocator strategy than random scan.

I actually prefer *net.IP instead of net.IP because reflect.Size(SliceHeader) > reflect.Size(uintptr).

Reply:
done.